org.eclipse.jdt.text.tests/src/org/eclipse/jdt/text/tests/folding/MarkdownJavadocFoldingTest.java

```diff
@@ -0,0 +1,124 @@
+package org.eclipse.jdt.text.tests.folding;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.List;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+import org.eclipse.jdt.testplugin.JavaProjectHelper;
+
+import org.eclipse.core.runtime.CoreException;
+
+import org.eclipse.jface.preference.IPreferenceStore;
+
+import org.eclipse.jface.text.IRegion;
+
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.IPackageFragment;
+import org.eclipse.jdt.core.IPackageFragmentRoot;
+
+import org.eclipse.jdt.ui.PreferenceConstants;
+import org.eclipse.jdt.ui.tests.core.rules.Java22ProjectTestSetup;
+import org.eclipse.jdt.ui.tests.core.rules.Java23ProjectTestSetup;
+import org.eclipse.jdt.ui.tests.core.rules.ProjectTestSetup;
+
+import org.eclipse.jdt.internal.ui.JavaPlugin;
+
+@RunWith(Parameterized.class)
+public class MarkdownJavadocFoldingTest {
+	@Rule
+	public ProjectTestSetup projectSetup;
+
+	private IJavaProject fJProject1;
+
+	private IPackageFragmentRoot fSourceFolder;
+
+	private IPackageFragment fPackageFragment;
+
+	private final boolean extendedFoldingActive;
+
+	private final boolean hasMarkdownJavadocs;
+
+	public MarkdownJavadocFoldingTest(boolean extendedFoldingActive, boolean hasMarkdownJavadocs) {
+		this.extendedFoldingActive= extendedFoldingActive;
+		this.hasMarkdownJavadocs= hasMarkdownJavadocs;
+		if (hasMarkdownJavadocs) {
+			projectSetup= new Java23ProjectTestSetup(false);
+		} else {
+			projectSetup= new Java22ProjectTestSetup();
+		}
+	}
+
+	@Parameters(name= "Extended folding active: {0}, JDK compilance with markdown Javadocs: {1}")
```

Comment:
What is the purpose of the first parameter?  The tests don't seem to do anything different based on whether the preference is set or not.